File: 10000번대/10800~10899/10845.py
class queue():
    last = 0
    array = [None]*10000
    index = -1

    def push(self,n):
        self.array[self.last] = n
        self.last += 1

    def getSize(self):
        return (self.last - (self.index+1) )

    def empty(self):
        if(self.last == self.index+1):
            return 1
        return 0

    def pop(self):
        if(self.empty()):
            return -1
        self.index +=1
        return self.array[self.index]

    def front(self):
        if(self.empty()):
            return -1
        return self.array[self.index+1]

    def back(self):
        if(self.empty()):
            return -1
        return self.array[self.last-1]


a = queue()
times = int(input())
for i in range(times):
    inputs = input().split()
    command = inputs[0]
    if command == "push":
        a.push(inputs[1])
    elif command == "pop":
        print(a.pop())
    elif command == 'front':
        print(a.front())
    elif command == 'size':
        print(a.getSize())
    elif command =='empty':
        print(a.empty())
    else:
        print(a.back())

File: 10000번대/10800~10899/10845.java
import java.util.Scanner;

class Queue{
	int last = 0;
	int[] array = new int[100000];
	int index = -1;
	public void push(int n) {
		this.array[this.last++] = n;
	}
	public int getSize() {
		return this.last - (this.index + 1);
	}

	public int empty() {
		if (this.last == this.index + 1) {
			return 1;
		}
		return 0;
	}
	public int pop() {
		if (this.empty()==1) {
			return -1;
		}
		return this.array[this.index++];
	}

	public int front() {
		if(this.empty()==1) {
			return -1;
		}
		return this.array[this.index+1];
	}
	public int back() {
		if(this.empty()==1) {
			return -1;
		}
		return this.array[this.last-1];
	}

}


public class Main {

	public static void main(String[] args) {
		Scanner  sc = new Scanner(System.in);
		Queue q = new Queue();
		int times = sc.nextInt();
		for(int i =0; i < times; i++) {
			String temp = sc.next();
			if(temp.equals("pop")) {
				System.out.println(q.pop());
			}
			else if (temp.equals("push")) {
				int temps = sc.nextInt();
				q.push(temps);
			}
			else if (temp.equals("size")) {
				System.out.println(q.getSize());
			}
			else if (temp.equals("empty")) {
				System.out.println(q.empty());
			}
			else if(temp.equals("front")){
				System.out.println(q.front());
			}
			else {
				System.out.println(q.back());
			}
		}


	}
}
